### DISCUSSION TITLE
TS0601_thermostat configure schedule

### DISCUSSION THREAD

**Discussion by soft2u** [Koenkk/zigbee2mqtt#15438] [Devices]
Hello all, 

What is the format of **workdays_schedule** and **holidays_schedule** ? I have tried  this format but with no success, at the end, the device is configured with default schedule
`[
        {"hour":6,"minute":0,"temperature":19},
        {"hour":8,"minute":0,"temperature":20},
        {"hour":18,"minute":0,"temperature":21},
        {"hour":20,"minute":30,"temperature":20},
        {"hour":22,"minute":0,"temperature":19},
        {"hour":23,"minute":30,"temperature":15}
    ]`

I'm using the latest version of zigbee2mqtt - 1.28.4

![image](https://user-images.githubusercontent.com/25981136/206399164-30bde7a9-4ec0-4deb-a005-bb7f773b22b0.png)

Thank you for your support

**Comment by SmileyMan** [comment]
Faced the same issue on format. Ended up sending Payload via NodeRED.

**Reply by KalleKal** [reply]
could you describe please how you are sending the raw payload?

**Comment by KalleKal** [comment]
In my version of Z2M the format is well described but it wont work anyway: the apply-button is always deactivated. Is there any way to find the javascript function in the source to check what input format is needed?

excerpt from the description:

> schedule
> Schedule will work with "auto" preset. In this mode, the device executes a preset week programming temperature time and temperature. Before using these properties, check `working_day` property. Each day can contain up to 10 segments. At least 1 segment should be defined. Different count of segments can be defined for each day, e.g., 3 segments for Monday, 5 segments for Thursday, etc. It should be defined in the following format: `hours:minutes/temperature`. Minutes can be only tens, i.e., 00, 10, 20, 30, 40, 50. Segments should be divided by space symbol. Each day should end with the last segment of 24:00. Examples: `04:00/20 08:30/22 10:10/18 18:40/24 22:50/19.5`; `06:00/21.5 17:20/26 24:00/18`. The temperature will be set from the beginning/start of one period and until the next period, e.g., `04:00/20 24:00/22` means that from 00:00 to 04:00 temperature will be 20 degrees and from 04:00 to 00:00 temperature will be 22 degrees.

**Reply by soft2u** [reply]
Have You tried to fill both fields?  workdays and holliday

**Comment by tomhampshire** [comment]
I can confirm that, using this method, only the "holidays" state gets updated and the "workdays" will not ever be updated.

**Comment by didier-guerin85** [comment]
Hello, this works but does not keep the values.
 
![Capture d’écran 2022-12-31 165132](https://user-images.githubusercontent.com/100363194/210148828-dbec19d6-2e92-4d2d-bf22-359f692339dc.png)

and
![Capture d’écran 2022-12-31 165344](https://user-images.githubusercontent.com/100363194/210148866-01a882a4-495b-426e-b200-23599646e0fc.png)

**Comment by didier-guerin85** [comment]
and i confirm that, using this method, only the "holidays" state gets updated and the "workdays" not be updated.
 "workdays_schedule": "06:00/19°C 08:00/17°C 11:30/17°C 12:30/17°C 17:30/19°C 22:00/17°C"
"holidays_schedule": "07:30/18°C 09:30/16°C 11:30/17°C 13:30/17°C 20:30/18°C 23:30/17°C",

**Comment by didier-guerin85** [comment]
for information. there is no Publishing 'set' 'workday_schedule' send in commands

Debug 2023-01-02 18:52:38Received MQTT message on 'zigbee2mqtt/0x804b50fffeeeb97b/set' with data '{"holidays_schedule":"07:30/18 09:30/16 11:30/17 13:30/16 20:30/18 23:30/17 ","week":"5+2","workdays_schedule":"06:30/18 08:30/16 11:30/18 13:30/16 17:30/18 23:30/17 "}'
Debug 2023-01-02 18:52:38**Publishing 'set' 'holidays_schedule' to '0x804b50fffeeeb97b'**
Debug 2023-01-02 18:52:39Publishing 'set' 'week' to '0x804b50fffeeeb97b'
Debug 2023-01-02 18:52:42Received Zigbee message from '0x804b50fffeeeb97b', type 'commandDataResponse', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,0,190],"type":"Buffer"},"datatype":2,"dp":2}],"seq":64264}' from endpoint 1 with groupID 0
Info 2023-01-02 18:52:42MQTT publish: topic 'zigbee2mqtt/0x804b50fffeeeb97b', payload

**Comment by didier-guerin85** [comment]
fonctionne en faisant 2 composite. 
Si meilleure solution?
Merci de me réondre
![Capture d’écran 2023-01-07 152329](https://user-images.githubusercontent.com/100363194/211155575-8c6c4546-5526-4e4a-8e40-ed6d26185d2a.png)

OU

![Capture d’écran 2023-01-07 152056](https://user-images.githubusercontent.com/100363194/211155576-9237d0fa-9968-490c-ab0c-cf5995efdfc6.png)
OU

![Capture d’écran 2023-01-07 151926](https://user-images.githubusercontent.com/100363194/211155577-e1376a37-4d05-4b4b-814e-3afab0cd6e62.png)

Ca fonctionne mais effacement des saisies horaires  après un rechargement de la page HTML. C'est dommage.

Si vous avez mieux et plus propre , je suis preneur.

Merci de me répondre.

**Comment by didier-guerin85** [comment]
![Capture d’écran 2023-01-07 153054](https://user-images.githubusercontent.com/100363194/211155769-5c5e14cf-361d-4ca5-8ea8-6e18999f54f2.png)

**Comment by didier-guerin85** [comment]
Sinon, tout fonctionne bien et plus précis  après modification de la sonde...
